### DISCUSSION TITLE
Exception when override the DataObject

### DISCUSSION THREAD

**Discussion by MyZik** [pimcore/pimcore#7948] [Q&A / Support]
Hi! I wrote the issue hier: https://github.com/pimcore/pimcore/issues/7945

I'm sorry for duplicate, but I still have a problem and have no more ideas how can I fix it. It's a clear project with one DataObject that I override in my custom class.
"it should be sufficient to load the dump & configs" -> it's working by all another projects with models override. 

An error will be thrown by composer install.
<img width="828" alt="Bildschirmfoto 2021-01-20 um 12 21 50" src="https://user-images.githubusercontent.com/12817320/105168827-f6091600-5b1a-11eb-96fd-1b80c98b602c.png">

`/app/config/config.yml`
```yaml
pimcore:
    models:
        class_overrides:
            'Pimcore\Model\DataObject\User': 'AppBundle\Model\DataObject\User'
```

My AppBundle/Model/DataObject/User class:

```php
<?php

declare(strict_types=1);

namespace AppBundle\Model\DataObject;

use Pimcore\Model\DataObject\ClassDefinition\Data\Password;
use Pimcore\Model\DataObject\User as BaseUser;
use Symfony\Component\Security\Core\User\UserInterface;

class User extends BaseUser implements UserInterface
{
    /**
     * @return null
     */
    public function getSalt()
    {
        return null;
    }

    /**
     * Trigger the hash calculation to remove the plain text password from the instance. This
     * is necessary to make sure no plain text passwords are serialized.
     *
     * @inheritDoc
     */
    public function eraseCredentials()
    {
        /** @var Password $field */
        $field = $this->getClass()->getFieldDefinition('password');
        $field->getDataForResource($this->getPassword(), $this);
    }

    public function getUsername()
    {
        return $this->getEmail();
    }
}
```

**Comment by MyZik** [comment]
So, the code is from documentation: https://pimcore.com/docs/pimcore/current/Development_Documentation/Development_Tools_and_Details/Security_Authentication/Authenticate_Pimcore_Objects.html

**Comment by NiklasBr** [comment]
I've seen this happen when the PSR-4 autoload cannot find the class, solved it by running `composer dump-autoload --optimize --classmap-authoritative --apcu` any time a new class is added.

**Comment by MyZik** [comment]
@NiklasBr thank you for your answer. Unfortunately it still doesn't work...
<img width="868" alt="Bildschirmfoto 2021-01-20 um 14 19 44" src="https://user-images.githubusercontent.com/12817320/105180381-cb26be00-5b2a-11eb-9189-1cadb5eac91d.png">

<img width="868" alt="Bildschirmfoto 2021-01-20 um 14 20 19" src="https://user-images.githubusercontent.com/12817320/105180393-cd891800-5b2a-11eb-9c31-7cfe147aca0a.png">

**Reply by SaschaHowe** [reply]
I have a similar issue. I boiled it down to this. 

The problem: `bin/console` doesn't run
Reason: one or more DataObjects are not available (not yet generated, fresh install, new feature)

Because `bin/console` isn't working you can't generate the class. 

This seems to be only a problem, when a generated class that is somehow (directly or via inheritance) used in a place where Symfony tries to store it in the ServiceContainer fails. Symfony can't load the non existing file and you can't even generate it because the console doesn't start. 

My current thinking is, that you have to add the generated DataObjects to version control and basically the suggestion here https://github.com/pimcore/demo/blob/10.x/.gitignore#L27 to ignore generated classes is not a good idea. 

Am I missing something?
